applications/RANSApplication/python_scripts/formulations/turbulence_models/scalar_turbulence_model_rans_formulation.py

```diff
@@ -0,0 +1,208 @@
+
+        This solves the variable given in self.GetSolvingVariable(), using element and conditions
+        having prefixes provided by self.GetElementNamePrefix() and self.GetConditionNamePrefix()
+
+        If wall functions are used, then self.GetConditionNamePrefix() should return non-empty prefix
+        otherwise it should be empty.
+
+        Args:
+            model_part (Kratos.ModelPart): ModelPart to be used in the formulation.
+            settings (Kratos.Parameters): Settings to be used in the formulation.
+        """
+        super().__init__(model_part, settings)
+
+        defaults = Kratos.Parameters(r"""{
+            "relative_tolerance"    : 1e-3,
+            "absolute_tolerance"    : 1e-5,
+            "max_iterations"        : 200,
+            "relaxation_factor"     : 0.5,
+            "echo_level"            : 2,
+            "linear_solver_settings": {
+                "solver_type"  : "amgcl"
+            },
+            "boundary_flags": ["INLET", "STRUCTURE"]
+        }""")
+
+        settings = self.GetParameters()
+        settings.ValidateAndAssignDefaults(defaults)
+        self.echo_level = settings["echo_level"].GetInt()
+
+    @abstractmethod
+    def GetSolvingVariable(self):
+        pass
+
+    @abstractmethod
+    def GetElementNamePrefix(self):
+        pass
+
+    @abstractmethod
+    def GetConditionNamePrefix(self):
+        pass
+
+    def PrepareModelPart(self):
+        self.turbulence_model_part = CreateRansFormulationModelPart(
+            self.GetComputingModelPart(),
+            self.__class__.__name__,
+            self.GetDomainSize(),
+            self.element_name,
+            self.condition_name)
+
+        Kratos.Logger.PrintInfo(self.__class__.__name__,
+                                "Created formulation model part.")
+
+    def Initialize(self):
+        InitializeYPlusVariablesInConditions(self.GetBaseModelPart())
+        CalculateNormalsOnConditions(self.GetBaseModelPart())
+
+        settings = self.GetParameters()
+
+        if (self.IsPeriodic()):
+            InitializePeriodicConditions(
+                self.GetBaseModelPart(),
+                self.GetModelPart(),
+                [self.GetSolvingVariable()])
+
+        linear_solver = GetKratosObjectType("LinearSolverFactory")(
+            settings["linear_solver_settings"])
+
+        builder_and_solver = CreateBlockBuilderAndSolver(
+            linear_solver,
+            self.IsPeriodic(),
+            self.GetCommunicator())
+
+        convergence_criteria = GetKratosObjectType("MixedGenericCriteria")([
+            (self.GetSolvingVariable(),
+             settings["relative_tolerance"].GetDouble(),
+             settings["absolute_tolerance"].GetDouble())])
+
+        if (self.is_steady_simulation):
+            scheme = self.scheme_type(settings["relaxation_factor"].GetDouble())
+        else:
+            scheme = GetKratosObjectType("BossakRelaxationScalarScheme")(
```

Comment:
I'd rename this variable to `scheme_type`. It's misleading.

Reply:
(same comment for the other variables)

Reply:
I am not sure I understand you correctly. `shceme` in here actually holds an object of `Scheme`. Why do you propose to change it to `shceme_type`?

Reply:
Then what is misleading is the `GetKratosObjectType` as well as the `scheme_type`. Shouldn't it be `GetKratosObject`?

Reply:
https://github.com/KratosMultiphysics/Kratos/blob/a35bf7bd8cbbaa6603182c1762ac59524520e1b9/applications/RANSApplication/python_scripts/formulations/turbulence_models/scalar_turbulence_model_rans_formulation.py#L102-L105

As per my understanding, `GetKratosObjectType("BossakRelaxationScalarScheme")` this gives the type of the object, next parenthesis gives you the constructor arguments to create an object of that type. It can be re-written in the following way which will work and will give the same result.

```python
 scheme_type = GetKratosObjectType("BossakRelaxationScalarScheme")
 scheme = scheme_type( 
     self.GetModelPart().ProcessInfo[Kratos.BOSSAK_ALPHA], 
     settings["relaxation_factor"].GetDouble(), 
     self.GetSolvingVariable()) 
```

Or did I understand it incorrectly? :/

Reply:
followed the second way as you suggested. :)